clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -17219,10 +17219,10 @@ static bool UsefulToPrintExpr(const Expr *E) {
   if (const auto *UnaryOp = dyn_cast<UnaryOperator>(E))
     return UsefulToPrintExpr(UnaryOp->getSubExpr());
 
-  // Ignore nested binary operators. This could be a FIXME for improvements
-  // to the diagnostics in the future.
-  if (isa<BinaryOperator>(E))
-    return false;
+  // Only print nested arithmetic operators.
+  if (const auto *BO = dyn_cast<BinaryOperator>(E))
+    return (BO->isShiftOp() || BO->isAdditiveOp() || BO->isMultiplicativeOp() ||
+            BO->isBitwiseOp());
 
```

Comment:
Maybe isComparisonOp too?

Reply:
Those are the ones we don't want to have, because we'd (ideally) have to show multiple comparisons and not just the one

Reply:
Can you clarify? I'm not sure i understood this comment. thanks

Reply:
For
```c++
static_assert(1 << 1 == 0);
```
we can print the result rather easily, but for 
```c++
static_assert(foo() > 1 && bar() == 10);
```
it's kind useless to print the LHS and RHS of the `&&` operator because they just evaluate to `true` or `false`. To print this in a useful way, we would have to dig deeper into the expression and print what `foo()` and `bar()` evaluate to.

Reply:
Oh, I see